### DISCUSSION TITLE
Twrnc + stylesheet

### DISCUSSION THREAD

**Discussion by Fgabz** [jaredh159/tailwind-react-native-classnames#328] [Q&A]
Hello,

I was wondering if using twrnc style directly inside Stylesheet.create has any benefit in terms of performance ? (Despite the fact having all the style for a component in one place)

Does twrnc can benefit from Stylesheet ?

Thanks !

**Comment by jaredh159** [comment]
I honestly don't know enough to say whether that would have any benefit. You could try benchmarking it, but twrnc tends to be very fast as it is, so i'm not sure you'd see much difference. Are you observing a performance issue you're trying to optimize? Or just wondering?

**Reply by Fgabz** [reply]
Not really (also complicated to know if perf are due to some styling issue or something else) but I'm more wondering. 
But my question comes from the assumption that:

- StyleSheet.create() transforms style objects into ID-based references
- These IDs are sent to the native thread once, during app initialization => **Does it mean that twrnc even if it's cached it's passing the full cache object to native bridge everytime ?**
- When using inline styles, new style objects are created on every render => **a bit the same question than above.**
- Each new object creation consumes memory and triggers garbage collection

**Reply by LouisLecouturier** [reply]
Hey @Fgabz, do you have any new insights on this ? Have you tried it ?

**Reply by Fgabz** [reply]
@LouisLecouturier Unfortunately I didn't not get the time. Let mw know if you are able to do it
